retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/BufferSinkHttpData.java

```diff
@@ -163,7 +165,7 @@ public boolean isEndOfStream() {
 
     @Override
     public byte[] array() {
-        return bytes;
+        return buffer.readByteArray();
```

Comment:
I introduced BufferSinkHttpData to reduce memory copy (https://github.com/line/armeria/pull/297#discussion_r81935874 ) but this change add one more memory copy. If we cannot this memory copy, it is better to use `Buffer` to simplify code and avoid yet another bug related to this class. 
@trustin what do you think?